### ISSUE TITLE
Transifex breaks sprintf format/dollar signs in translations

### ISSUE THREAD

**Issue by angelorocha** [Elgg/Elgg#7540]
Change lang file name of portuguese files to "pt_br" (currently this as pt_BR). for this reason the translation is not working.

**Comment by mrclay** [comment]
Strawman: We rewrite (var_export) all lang files to use `"` delimiters. We make `Elgg\I18n\Translator::getAllLanguageCodes` static and use it to filter files coming from Transifex.

**Comment by ewinslow** [comment]
IIUC, this is breaking in 1.x, too, so its not a 2.x blocker, since it's not a regression.

Tenatively removing from 2.x milestone. Please do holler if I'm mistaken here somehow.

**Comment by skhiro** [comment]
I have already fixed on Transifex . Upload and marge to github please.

**Comment by ewinslow** [comment]
@skhiro when did you fix it? I recently released a new version of 1.12 and 2.0, so it's possible the change  has already been included.

**Comment by skhiro** [comment]
@ewinslow  Immediately after #8561. about one month ago.
Please check engine and web_services in Transifex.

**Comment by ewinslow** [comment]
There are still a couple of strings there that seem to be problematic. New
translations are pulled down automatically at each release. Is it possible
that just changing the quote didn't trigger the download?

**Comment by skhiro** [comment]
There are difference between github and Transifex and issues are still remained in github japanese translagions.
 * languages/ja.php and engine
 * mod/web_services/languages/ja.php and web_services

It seems changes are not updated automatically. I have no idea how and what  does trigger updating in Transifex( or github).

**Comment by ewinslow** [comment]
:( are you familiar with how to do a pull request? If so you could submit
one of those? If not, perhaps you could compile a complete list of line
numbers that need to be fixed and I can submit that PR. Doesn't fix the
problem permanently but I want to get something in so people can enjoy the
fruits of your labors!

**Comment by skhiro** [comment]
@ewinslow I don't know much about PR,  please PR for me.
Changes are below.

in languages/ja.php
line 106:
```
-       'view:missing_param' => "View %2$s で必要なパラメータ「 %1$s 」がありません。",
+       'view:missing_param' => "View %2\$s で必要なパラメータ「 %1\$s 」がありません。",
```
in mod/web_services/languages/ja.php
line 4:
```
-       'APIException:MissingParameterInMethod' => "メソッド「 %2$s 」のパラメタ「 %1$s 」が欠落
しています",
+       'APIException:MissingParameterInMethod' => "メソッド「 %2\$s 」のパラメタ「 %1\$s 」が欠
落しています",
```
line 6:
```
-       'APIException:UnrecognisedTypeCast' => "メソッド「 %3$s 」の変数「 %2$s 」のキャスト %1$s の型がわかりません。",
+       'APIException:UnrecognisedTypeCast' => "メソッド「 %3\$s 」の変数「 %2\$s 」のキャスト %1\$s の型がわかりません。",
```
line 7:
```
-       'APIException:InvalidParameter' => "メソッド「 %2$s 」のパラメタ「 %1$s 」が不適切な値で
した。",
+       'APIException:InvalidParameter' => "メソッド「 %2\$s 」のパラメタ「 %1\$s 」が不適切な値
でした。",
```
line 32:
```
-       'InvalidParameterException:UnrecognisedHttpMethod' => "APIメソッド「 %2$s 」を呼び出すた
めの httpメソッド %1$s が認識されませんでした",
+       'InvalidParameterException:UnrecognisedHttpMethod' => "APIメソッド「 %2\$s 」を呼び出す
ための httpメソッド %1\$s が認識されませんでした",
```

**Comment by mrclay** [comment]
Putting this on 2.0 milestone while mod/embed it.php is broken

**Comment by mrclay** [comment]
I made a separate issue for PHP syntax errors. Sorry for the noise.
